chatwoot/chatwoot
# feat: show alert if billing is pending

Depends On: https://github.com/chatwoot/chatwoot/pull/7296

---

Show billing pending banner if payment is due for an account

This PR has the following changes

1. Update Stripe webhook event to include subscription status and cycle end date
2. Update banner design to have better buttons
3. Add payment pending banner
4. Add limits exceeded banner

## Type of change


## How Has This Been Tested?

Tested locally with stripe CLI.

<details>
<summary>Screenshot: Pending Payment</summary>

<img width="1512" alt="CleanShot 2023-06-05 at 18 14 26@2x" src="https://github.com/chatwoot/chatwoot/assets/18097732/a29e0c12-d239-423f-a76b-24eccc9cd490">

</details>

<details>
<summary>Screenshot: Limits exceeded</summary>

<img width="1512" alt="CleanShot 2023-06-08 at 16 24 25@2x" src="https://github.com/chatwoot/chatwoot/assets/18097732/1b4343e1-c1f4-4b9a-a0ae-7d2419a79c7f">

</details>



### How to test

1. Update the `custom_attributes` for an account and add the following `"subscription_status": "past_due",
  "subscription_ends_on": "2023-06-02T08:21:57.000Z"`
2. Open the UI and see if the banner appears
3. Dismissing the banner should hide it for 24 hours for that browser
5. Open billing should take them to the billing page

> note: this works if the user is admin and `isOnChatwootCloud` is `true`

### Banner Component Preview

<details>
<summary>Updated banner component in the UI</summary>

#### Assign yourself

<img width="1512" alt="CleanShot 2023-06-08 at 16 26 48@2x" src="https://github.com/chatwoot/chatwoot/assets/18097732/7f8c9e62-df8a-45b4-8b93-585b077ac7ad">

#### Select a tweet to reply

<img width="1512" alt="CleanShot 2023-06-02 at 15 32 35@2x" src="https://github.com/chatwoot/chatwoot/assets/18097732/bb67b191-26c1-4971-a4e4-1aafa6d6e70b">

#### cannot reply

<img width="1512" alt="CleanShot 2023-06-02 at 15 32 16@2x" src="https://github.com/chatwoot/chatwoot/assets/18097732/57f4d49e-0739-40b2-ba46-aa0adb3feb5b">

</details>

## Checklist:
